stdlib/tkinter/__init__.pyi

```diff
@@ -384,8 +389,32 @@ class Misc:
     @overload
     def grid_bbox(self, column: int, row: int, col2: int, row2: int) -> Tuple[int, int, int, int] | None: ...
     bbox = grid_bbox
-    def grid_columnconfigure(self, index, cnf=..., **kw): ...  # TODO
-    def grid_rowconfigure(self, index, cnf=..., **kw): ...  # TODO
+    # The cnf parameter's default value is an empty dict. That's a perfectly
+    # valid instance of _GridIndexInfo, but MyPy declares that an empty dict
+    # literal must be dict[Any, Any] and is therefore incompatible.
+    #
+    # The return value is always _GridIndexInfo if no optional parameters are
+    # provided, and always None if one or more are provided. Unfortunately,
+    # type inference requires both cases have the same return type, so we must
+    # combine them.
+    def grid_columnconfigure(
+        self,
+        index: _GridIndex,
+        cnf: _GridIndexInfo = ...,  # type: ignore
+        minsize: _ScreenUnits = ...,
+        pad: _ScreenUnits = ...,
+        uniform: str | None = ...,
+        weight: int = ...,
+    ) -> _GridIndexInfo | None: ...
```

Comment:
This is a bit annoying to use in practice, because you need to do this if you want to access the returned `_GridIndexInfo`:

```python
info = widget.grid_columnconfigure(index)`
assert info is not None
print(info["minsize"])
```

You can instead do this:

```suggestion
    ) -> _GridIndexInfo | Any: ...  # can be None but annoying to check
```

Unlike you would expect, this is not same as returning `Any`, and it works as intended:
- You don't need to `assert info is not None`, because `info["minsize"]` is valid when `info` has type `_GridIndexInfo` or type `Any`.
- You will still get an error for `info["lolwat"]`, because to not get an error, it would have to be valid for both union members but it is not for `_GridIndexInfo`.

Also, the one-liner comment I suggested above is enough, and we no longer need a wall of text. Typeshed maintainers know these tricks, and a short reminder is enough.

Reply:
Ah, just what I wanted! Thanks!